iree/compiler/Conversion/Common/LinalgBufferizePass.cpp

```diff
@@ -197,7 +199,19 @@ static LogicalResult convertAnyLinalgOp(
   SmallVector<Value, 2> newOutputBuffers;
   if (failed(allocateBuffersForResults(b, loc, allocationFn, op,
                                        newOutputBuffers, bvm))) {
-    assert(false);
+    LLVM_DEBUG(llvm::dbgs()
+               << "failed to allocate output buffers for op: " << op << "\n");
+    return failure();
+  }
+
+  // If this is a fill op that acts as a tensor initializer to other linalg ops,
+  // check whether the initialization is used. If not we can just drop the op.
+  if (auto fillOp = dyn_cast<linalg::FillOp>(op.getOperation())) {
```

Comment:
This seems like the wrong place to do this. Its effectively a dead op? This should move upstream. At least not during bufferization. Its already fairly complex.

Reply:
Hmm, looks like this is not needed now; maybe because some recent landed changes since I've this implemented. I'll remove this.

But good point. I'll still send this out as a general canonicalization pattern upstream in Linalg.